app/commands/xp.js

```diff
@@ -0,0 +1,444 @@
+                { text: `You have retrieved `, color: 'white' },
+                { text: `${pointsRetrievedAll}`, color: 'red' },
+                { text: ` experience points.\nYou have `, color: 'white' },
+                { text: `0`, color: 'red' },
+                { text: ` points remaining.`, color: 'white' },
+            ]);
+
+        } else {
+            //make args[0] an integer
+            let argsInt = parseInt(args[0], 10);
+            
+            if (argsInt >= 0) {
+                //need to handle when player inputs more xp than they have xp stored
+                if (argsInt <= currentBalance) {
+                    //input less than or equal the xp stored.
+
+                    //adding the xp equal to number input as argsInt
+                    let pointsRetrievedPartial = await this.addPlayerExperience(playerName, argsInt);
+
+                    //removing those points from their xp store
+                    await this.updatePlayerXpStore(playerName, currentBalance - pointsRetrievedPartial);
+
+                    //informing player current point balance
+                    this.tellPlayerRaw(playerName, [
+                        { text: `You have retrieved `, color: 'white' },
+                        { text: `${pointsRetrievedPartial}`, color: 'red' },
+                        { text: ` experience points.`, color: 'white' },
+                    ]);
+                    await this.simpleCheck(playerName);
+
+                    
+                } else {
+                    //input too much xp.
+
+                    //just retrieving all thier xp.
+                    let pointsRetrievedInsufficient = await this.addPlayerExperience(playerName, currentBalance);
+
+                    //removing those points from their xp store
+                    await this.updatePlayerXpStore(playerName, currentBalance - pointsRetrievedInsufficient);
+
+                    //informing player current point balance
+                    this.tellPlayerRaw(playerName, [
+                        { text: `Not enough stored experience points.\nRetrieving `, color: 'white' },
+                        { text: `${pointsRetrievedInsufficient}`, color: 'red' },
+                        { text: ` experience points instead.`, color: 'white' },
+                    ]);
+                    await this.simpleCheck(playerName);
+                }
+
+            } else {
+                //They got here because they messed up
+                this.tellPlayerRaw(playerName, [
+                    { text: `Must input a positive number.\n`, color: 'red' },
+                    { text: `!xp get 50`, color: 'green' },
+                    { text: ` would retrieve 50 experience points.`, color: 'red' },
+                ]);
+            }
+        }
+    }
+
+    async handleXpGive(playerName, args) {
+        let loggedInPlayers = await this.getListOfOnlinePlayers();
+
+        if (args[0] && args[0].toLowerCase() === 'accept') {
```

Comment:
This is an excellent time to use the return statement in an if block if you did something like this

`if (args[0] && args[0].toLowerCase() === 'accept') return this.handleXpGiveAccept(playerName);`

You could lower the indent of everything following make it easier to read